internal/events/eventer_test.go

```diff
@@ -60,6 +61,11 @@ func fakeHandler(id string, out chan eventPair) events.Handler {
 		return nil
 	}
 }
+
+func alwaysFailingHandler(_ *message.Message) error {
+	return errors.New("handler always fails")
```

Comment:
It would be nice to keep a counter in here so we can verify this got called the right number of times, to test that retry is working.

Reply:
You can do this with a function that returns a function:

```golang
func countFailuresHandler(counter *int) events.Handler {
	return func(_ *message.Message) error {
		counter++
		return errors.New("handler always fails)
	}
}
```

(and then rig up a counter and a `wantsCalls` in the test.)

Feel free to defer to a separate PR if you want.